packages/beacon-node/src/network/processor/index.ts

```diff
@@ -123,13 +132,14 @@ export class NetworkProcessor {
     let jobsSubmitted = 0;
 
     job_loop: while (jobsSubmitted < MAX_JOBS_SUBMITTED_PER_TICK) {
-      // Check canAcceptWork before calling queue.next() since it consumes the items
-      if (!this.chain.blsThreadPoolCanAcceptWork() || !this.chain.regenCanAcceptWork()) {
-        this.metrics?.networkProcessor.canNotAcceptWork.inc();
-        break;
-      }
-
       for (const topic of executeGossipWorkOrder) {
+        // Check canAcceptWork before calling queue.next() since it consumes the items
+        // beacon block is guaranteed to be processed immedately
+        const bypassQueue = executeGossipWorkOrderObj[topic]?.bypassQueue ?? false;
+        if (!bypassQueue && (!this.chain.blsThreadPoolCanAcceptWork() || !this.chain.regenCanAcceptWork())) {
```

Comment:
This will cause to have to call canAcceptWork many more times. Why not:

```ts
const canAcceptWork = this.chain.blsThreadPoolCanAcceptWork() && this.chain.regenCanAcceptWork()

for (const topic of executeGossipWorkOrder) {
        // Check canAcceptWork before calling queue.next() since it consumes the items
        // beacon block is guaranteed to be processed immedately
        const bypassQueue = executeGossipWorkOrderObj[topic]?.bypassQueue ?? false;
        if (!executeGossipWorkOrderObj[topic].bypassQueue && !canAcceptWork) {
           break job_loop;
        }
```

Reply:
good point 👍 , I modified accordingly